packages/language-support/src/antlr-grammar/CypherParser.g4

```diff
@@ -1553,16 +1602,22 @@ symbolicNameOrStringParameter
    | parameter["STRING"]
    ;
 
+aliasName:
+    symbolicAliasNameOrParameter;
+
+databaseName:
+    symbolicAliasNameOrParameter;
```

Comment:
The change I was expecting was to separate into:

```
createAliasOrDatabaseName: 
    symbolicAliasNameOrParameter;

ReferenceAliasorDatabaseName:
    symbolicAliasNameOrParameter;
```

Because then we could use `ReferenceAliasorDatabaseName` instead of `RULE_symbolicAliasName` in the `preferredRules` and wouldn't have to check the tree if it's a rule that creates or references an alias/database. It could still be worth splitting database & aliases as well and this change also solves the problem, so I'm fine doing it this way if you prefer

Reply:
Yeah I agree that would be nicer for us but it could make it more difficult to enforce for engineers working in the database because it obfuscates the grammar a step further.

I'm happy to keep evolving the grammar, but we should make it gradually because pushing any change is a pain in the monorepo. Out of curiosity I completely tried to split database completions from alias completions here: https://github.com/neo4j/cypher-language-support/commit/245b807a2848c6cee1da1dab2468cef3df223c3d and it was super painful, so I decided to revert it 😬

Reply:
alright, good to hear it was considered!